tests/test_custom_models.py

```diff
@@ -444,6 +457,128 @@ def test_repr_lora_conv2d(self):
         self.assertTrue("default" in print_output)
 
 
+class MultipleActiveAdaptersTester(unittest.TestCase):
+    """
+    A test class to test the functionality of multiple active adapters.
+
+    This is not specifically tied to custom models, it's just easy to test here and testing it on all types of models
+    would be overkill.
+    """
+
+    def setUp(self):
+        super().setUp()
+        self.configs = {
+            "lora": {"class": LoraConfig, "kwargs": {"target_modules": ["lin0"], "init_lora_weights": False}},
+            "ia3": {
+                "class": IA3Config,
+                "kwargs": {
+                    "target_modules": ["lin0"],
+                    "feedforward_modules": ["lin0"],
+                    "init_ia3_weights": False,
+                },
+            },
+            "adalora": {
+                "class": AdaLoraConfig,
+                "kwargs": {"target_modules": ["lin0"], "init_lora_weights": False, "inference_mode": True},
+            },
+        }
+
+    def prepare_inputs_for_testing(self):
+        X = torch.arange(90).view(9, 10)
+        return {"X": X}
+
+    def get_adapter_config(self, tuner_method, targets):
+        if targets == "same":
+            return self.configs[tuner_method]
+        config = copy.deepcopy(self.configs[tuner_method])
+        kwargs = config["kwargs"]
+        for key, value in kwargs.items():
+            new_targets = []
+            if not isinstance(value, list):
+                continue
+            for target in value:
+                new_targets.append("lin1" if target == "lin0" else "lin0")
+            kwargs[key] = new_targets
+        return config
+
+    def set_multiple_active_adapters(self, model, adapter_names):
+        for module in model.modules():
```

Comment:
Oh, we should add support for `lora_model.set_adapter(["adapter1", "adapter2"])` etc. Quite annoying for users if they have to do this themselves.

Reply:
Yes, let's take that up in a separate PR